keps/sig-storage/3756-volume-reconstruction/README.md

```diff
@@ -0,0 +1,905 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+After kubelet is restarted, it looses track of all volume it mounted for
+running Pods. It tries to restore this state from the API server, where kubelet
+can find Pods that _should_ be running, and from the host's OS, where it can
+find actually mounted volumes. We know this process is imperfect.
+This KEP tries to rework the process. While the work is technically a bugfix,
+it changes large parts of kubelet, and we'd like to have it behind a feature
+gate to provide users a way to get to the old implementations in case of
+problems.
+
+This work started as part of
+[KEP 1790](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1710-selinux-relabeling)
+and even went alpha in v1.26, but we'd like to have a separate feature + feature
+gate to be able to graduate VolumeManager reconstruction faster.
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+### Goals
+
+* During kubelet startup, allow it to populate additional information about
+  _how_ are existing volumes mounted.
+  [KEP 1710](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1710-selinux-relabeling)
+  needs to know what mount options did the previous kubelet used when mounting
+  the volumes, to be able to tell if they need any change or not.
+* Fix [#105536](https://github.com/kubernetes/kubernetes/issues/105536): Volumes
```

Comment:
Is https://github.com/kubernetes/kubernetes/issues/111933 the same issue or different?

Reply:
It looks to me the same to me.
BTW, I am not sure if the new reconstruction actually helps with the issue - to me it seems that the reconstruction itself failed and it is not retried in this KEP either.